e2e/test/scenarios/custom-column/custom-column-reproductions.cy.spec.js

```diff
@@ -1055,69 +1053,83 @@ describe("issue 49882", () => {
+    // instead.
+
     H.enterCustomColumnDetails({
-      formula: `case([Tax] > 1, case([Total] > 200, [Total], "Nothing"), [Tax])${selectTax}`,
+      formula:
+        'case([Tax] > 1, case([Total] > 200, [Total], "Nothing"), [Tax])',
+      blur: false,
+      allowFastSet: true,
     });
-    cy.get(".ace_text-input").first().focus().realPress(["Control", "X"]);
-    cy.get(".ace_text-input")
-      .first()
-      .focus()
-      .type(moveCursorBefore2ndCase)
-      .realPress(["Control", "V"]);
-    cy.get(".ace_text-input").first().focus().type(" ").blur();
-
-    cy.get(".ace_text-input")
-      .first()
-      .focus()
-      .should(
-        "have.value",
-        'case([Tax] > 1, [Tax] case([Total] > 200, [Total], "Nothing"), )\n\n',
-      );
+
+    // Cut [Tax]
+    H.CustomExpressionEditor.type(
+      'case([Tax] > 1, case([Total] > 200, [Total], "Nothing"), )',
+      {
+        allowFastSet: true,
+      },
+    );
+
+    // Paste [Tax] before case
+    H.CustomExpressionEditor.type(
+      'case([Tax] > 1, [Tax] case([Total] > 200, [Total], "Nothing"), )',
+      {
+        allowFastSet: true,
+      },
+    );
+
+    H.CustomExpressionEditor.shouldContain(
+      'case([Tax] > 1, [Tax] case([Total] > 200, [Total], "Nothing"), )',
+    );
+    H.CustomExpressionEditor.blur();
 
     H.popover()
       .findByText("Expecting comma but got case instead")
       .should("be.visible");
   });
 
-  it("should allow moving cursor between wrapped lines with arrow up and arrow down keys (metabase#49882-3)", () => {
+  // TODO: we no longer have wrapped lines (for now)
+  it.skip("should allow moving cursor between wrapped lines with arrow up and arrow down keys (metabase#49882-3)", () => {
     H.enterCustomColumnDetails({
       formula:
         'case([Tax] > 1, case([Total] > 200, [Total], "Nothing"), [Tax]){leftarrow}{leftarrow}{uparrow}x{downarrow}y',
     });
 
-    cy.get(".ace_text-input")
-      .first()
-      .should(
-        "have.value",
-        'case([Tax] > 1, xcase([Total] > 200, [Total], "Nothing"), [Tax]y)\n\n',
-      );
+    H.CustomExpressionEditor.shouldContain(
+      'case([Tax] > 1, xcase([Total] > 200, [Total], "Nothing"), [Tax]y)',
+    );
   });
 
   it("should update currently selected suggestion when suggestions list is updated (metabase#49882-4)", () => {
     const selectProductVendor =
-      "{downarrow}{downarrow}{downarrow}{downarrow}{downarrow}{enter}";
+      "{downarrow}{downarrow}{downarrow}{downarrow}{downarrow}";
     H.enterCustomColumnDetails({
-      formula: `[Product${selectProductVendor}{leftarrow}{leftarrow}`,
+      formula: `[Produ${selectProductVendor}`,
       blur: false,
     });
 
-    cy.findByTestId("expression-suggestions-list-item")
-      .should("have.text", "Product → Vendor")
-      .and("have.css", "background-color", "rgb(80, 158, 227)");
+    H.CustomExpressionEditor.completion("Product → Rating").should(
+      "be.visible",
+    );
```

Comment:
Let's not remove the assertion that verifies the suggestion is selected

Reply:
That's what the lines below are doing: accepting the completion should result in the selected completion being used as the value.